query-connector/src/app/fhir-servers/page.tsx

```diff
@@ -0,0 +1,392 @@
+"use client";
+import { Icon, Label, TextInput } from "@trussworks/react-uswds";
+import {
+  deleteFhirServer,
+  getFhirServerConfigs,
+  insertFhirServer,
+  updateFhirServer,
+} from "../database-service";
+import SiteAlert from "../query/designSystem/SiteAlert";
+import Table from "../query/designSystem/table/Table";
+import { useEffect, useState, useRef } from "react";
+import { FhirServerConfig } from "../constants";
+import { Modal, ModalRef } from "../query/designSystem/modal/Modal";
+import styles from "./fhirServers.module.scss";
+import classNames from "classnames";
+
+type ModalMode = "create" | "edit";
+
+/**
+ * Client side parent component for the FHIR servers page. It displays a list of FHIR servers
+ * @returns - The FhirServers component.
+ */
+const FhirServers: React.FC = () => {
+  const [fhirServers, setFhirServers] = useState<FhirServerConfig[]>([]);
+  const [serverName, setServerName] = useState("");
+  const [serverUrl, setServerUrl] = useState("");
+  const [connectionStatus, setConnectionStatus] = useState<
+    "idle" | "success" | "error"
+  >("idle");
+  const [errorMessage, setErrorMessage] = useState<string | undefined>("");
+  const [selectedServer, setSelectedServer] = useState<FhirServerConfig | null>(
+    null,
+  );
+  const [modalMode, setModalMode] = useState<ModalMode>("create");
+  const modalRef = useRef<ModalRef>(null);
+
+  useEffect(() => {
+    getFhirServerConfigs(true).then((servers) => {
+      setFhirServers(servers);
+    });
+  }, []);
+
+  const resetModalState = () => {
+    setServerName("");
+    setServerUrl("");
+    setConnectionStatus("idle");
+    setErrorMessage("");
+    setSelectedServer(null);
+  };
+
+  const handleOpenModal = (mode: ModalMode, server?: FhirServerConfig) => {
+    setModalMode(mode);
+    if (mode === "edit" && server) {
+      setSelectedServer(server);
+      setServerName(server.name);
+      setServerUrl(server.hostname);
+    } else {
+      resetModalState();
+    }
+    modalRef.current?.toggleModal();
+  };
+
+  const handleCloseModal = () => {
+    resetModalState();
+    modalRef.current?.toggleModal();
+  };
+
+  interface ConnectionTestResult {
+    success: boolean;
+    error?: string;
+  }
+
+  /**
+   * Tests a connection to a FHIR server
+   * @param url - The URL of the FHIR server to test
+   * @returns A promise that resolves to an object containing the test result
+   */
+  const testFhirConnection = async (
```

Comment:
Definitely could be a separate ticket/post-connectathon but I think we should try to test the Connection upon insertion, at least for the default servers, so that when a user first looks at the table the status is not null.

Reply:
It actually already does this! See line 178